# DNA
Rosalind algorithmic DNA challenges
My python solutions for the Rosalind challenge, might be interesting for
some people to see how I solved those, please don't use it to cheat.
Will add some notes on what I found interesting while discovering the field.

# Required
- Python 3
- Biopython library
- requests library

# Links
- http://www.bioinformatics.babraham.ac.uk/projects/download.html
- MEME suite http://meme-suite.org/index.html
- ORF Finder http://www.bioinformatics.org/sms2/orf_find.html
- EMBOSS http://emboss.sourceforge.net/what/
- LALIGN https://www.ebi.ac.uk/Tools/psa/lalign/

# Deeply studied organism
- E.coli bacteria: http://ecoliwiki.net
- C.elegans worm: http://www.wormbook.org/

# DNA Methylation
https://en.wikipedia.org/wiki/DNA_methylation
- C & A can be methylated.
- Strengthens this specific pair.
- Blocks
- Similar to escaping SQL filters.

# Mitochrondria
https://en.wikipedia.org/wiki/Mitochondrion

# Synteny Blocks
- Synteny blocks appear close to coding blocks or functions.
- Synteny blocks are very similar areas of two species genomes that have been flipped and moved around by rearrangements.

# KEGG pathways
- KEGG pathways are fascinating.

# Transposons (Nobel83)
- Different regions of the genome evolve in different ways. In coding regions, where any significant change can be lethal to the organism, the most common source of variation is point mutations. In non-coding regions (introns or intergenic spacers), we find a different situation: entire intervals can easily be duplicated, inserted, deleted, or reversed.
- Transposons are jumping genes in non-coding DNA
- Mobility of "variables" non-coding transposons
- Inter specie transposons is fascinating. https://en.wikipedia.org/wiki/Tn10
