# DNA
Rosalind algorithmic DNA challenges
My python solutions for the Rosalind challenge, might be interesting for
some people to see how I solved those, please don't use it to cheat.
Will add some notes on what I found interesting while discovering the field.

# Required
- Python 3
- Biopython library
- requests library

# Links
- http://www.bioinformatics.babraham.ac.uk/projects/download.html
- MEME suite http://meme-suite.org/index.html
- ORF Finder http://www.bioinformatics.org/sms2/orf_find.html
- EMBOSS http://emboss.sourceforge.net/what/
- LALIGN https://www.ebi.ac.uk/Tools/psa/lalign/

# Deeply studied organism
- E.coli bacteria: http://ecoliwiki.net
- C.elegans worm: http://www.wormbook.org/
- Virii: https://viralzone.expasy.org/

# Multiple reading frames
- The usage of multiple reading frames leads to the possibility of overlapping genes; there may be many of these in virus, prokaryote, and mitochondrial genomes.[7] Some viruses, e.g. Hepatitis B virus and BYDV, use several overlapping genes in different reading frames.

# DNA Methylation
https://en.wikipedia.org/wiki/DNA_methylation
- C & A can be methylated.
- Strengthens this specific pair.
- Blocks
- Similar to escaping SQL filters.

# Mitochrondria
https://en.wikipedia.org/wiki/Mitochondrion

# Synteny Blocks
- Synteny blocks appear close to coding blocks or functions.
- Synteny blocks are very similar areas of two species genomes that have been flipped and moved around by rearrangements.

# KEGG pathways
- KEGG pathways are fascinating.

# Point mutations
Because transversions require a more drastic change to the base's chemical structure, they are less common than transitions. Across the entire genome, the ratio of transitions to transversions is on average about 2. However, in coding regions, this ratio is typically higher (often exceeding 3) because a transition appearing in coding regions happens to be less likely to change the encoded amino acid, particularly when the substituted base is the third member of a codon (feel free to verify this fact using the DNA codon table). Such a substitution, in which the organism's protein makeup is unaffected, is known as a silent substitution.

# Transposons (Nobel83)
- Different regions of the genome evolve in different ways. In coding regions, where any significant change can be lethal to the organism, the most common source of variation is point mutations. In non-coding regions (introns or intergenic spacers), we find a different situation: entire intervals can easily be duplicated, inserted, deleted, or reversed.
- Transposons are jumping genes in non-coding DNA
- Mobility of "variables" non-coding transposons
- Inter specie transposons is fascinating. https://en.wikipedia.org/wiki/Tn10

# Observations
- Controlled environment in the scientific method destroy the ecosystem complexity.
